### PR TITLE
Bump Cairo compatibility

### PR DESCRIPTION
**Pull request by CiaranOMara**
This pull request sets Cairo compatibility to 0.7 and addresses associated deprecated bindings.

This request may be all that is required to address issue #357.

### CODE REVIEW

**Comment by Mattriks** [comment]
Have you tested this with a julia unicode example? e.g.
```julia
# the string is \ita \itb etc
img = compose(context(), 
    text(0.5, 0.5, "𝑎 𝑏 𝑐 𝑑 𝑒 𝑓", hcenter, vcenter), font("Segoe UI"), fontsize(20pt) )
draw(PNG(), img)
```
This works for me on Cairo 0.6, but only shows boxes with Cairo 0.7 (on windows 10 OS).

**Comment by CiaranOMara** [comment]
That type of example does not appear to be included in a current integration tests.

Below is the output of what I got for the above example with Cairo v0.7 on macOS 10.14.6. I can't get Cairo v0.6 to work.

```julia
julia> using Compose
julia> import Cairo, Fontconfig
julia> img = compose(context(), text(0.5, 0.5, "𝑎 𝑏 𝑐 𝑑 𝑒 𝑓", hcenter, vcenter), font("Segoe UI"), fontsize(20pt) )
Context(Measures.BoundingBox{Tuple{Measures.Length{:w,Float64},Measures.Length{:h,Float64}},Tuple{Measures.Length{:w,Float64},Measures.Length{:h,Float64}}}((0.0w, 0.0h), (1.0w, 1.0h)), nothing, nothing, nothing, nothing, List([]), List([Compose.Form{Compose.TextPrimitive{Tuple{Measures.Length{:cx,Float64},Measures.Length{:cy,Float64}},Rotation{Tuple{Measures.Length{:w,Float64},Measures.Length{:h,Float64}}},Tuple{Measures.Length{:mm,Float64},Measures.Length{:mm,Float64}}}}(Compose.TextPrimitive{Tuple{Measures.Length{:cx,Float64},Measures.Length{:cy,Float64}},Rotation{Tuple{Measures.Length{:w,Float64},Measures.Length{:h,Float64}}},Tuple{Measures.Length{:mm,Float64},Measures.Length{:mm,Float64}}}[Compose.TextPrimitive{Tuple{Measures.Length{:cx,Float64},Measures.Length{:cy,Float64}},Rotation{Tuple{Measures.Length{:w,Float64},Measures.Length{:h,Float64}}},Tuple{Measures.Length{:mm,Float64},Measures.Length{:mm,Float64}}}((0.5cx, 0.5cy), "𝑎 𝑏 𝑐 𝑑 𝑒 𝑓", Compose.HCenter(), Compose.VCenter(), Rotation{Tuple{Measures.Length{:w,Float64},Measures.Length{:h,Float64}}}(0.0, (0.5w, 0.5h)), (0.0mm, 0.0mm))], Symbol(""))]), List([Compose.Property{Compose.FontSizePrimitive}(Compose.FontSizePrimitive[Compose.FontSizePrimitive(7.055555555555555mm)]), Compose.Property{Compose.FontPrimitive}(Compose.FontPrimitive[Compose.FontPrimitive("Segoe UI")])]), 0, false, false, false, false, nothing, nothing, 0.0, Symbol(""))

julia> draw(PNG(), img)
Compose.Image{Compose.PNGBackend}(IOBuffer(data=UInt8[...], readable=true, writable=true, seekable=true, append=false, size=2758, maxsize=Inf, ptr=2759, mark=-1), Cairo.CairoSurfaceBase{UInt32}(Ptr{Nothing} @0x0000000000000000, 535.0, 378.0), Cairo.CairoContext(Ptr{Nothing} @0x0000000000000000, Cairo.CairoSurfaceBase{UInt32}(Ptr{Nothing} @0x0000000000000000, 535.0, 378.0), Ptr{Nothing} @0x00007fd12118b2e0), 534.5059133378628, 377.9527559055118, RGBA{Float64}(0.0,0.0,0.0,0.0), RGBA{Float64}(0.0,0.0,0.0,1.0), Float64[], LineCapButt(), Compose.LineJoinMiter(), true, 0.3mm, 3.880555555555555mm, "Helvetica Neue,Helvetica,Arial,sans", nothing, false, Compose.ImagePropertyState[], Compose.ImagePropertyFrame[], Dict{Type,Union{Nothing, Compose.Property}}(), true, false, nothing, true, true, 3.7795275590551185, nothing, nothing)
false

julia> draw(PNG("tmp.png"), img)
```
tmp


However, plain text is fine.
```julia
julia> img_plain = compose(context(), text(0.5, 0.5, "plain text", hcenter, vcenter), font("Segoe UI"), fontsize(20pt) )
Context(Measures.BoundingBox{Tuple{Measures.Length{:w,Float64},Measures.Length{:h,Float64}},Tuple{Measures.Length{:w,Float64},Measures.Length{:h,Float64}}}((0.0w, 0.0h), (1.0w, 1.0h)), nothing, nothing, nothing, nothing, List([]), List([Compose.Form{Compose.TextPrimitive{Tuple{Measures.Length{:cx,Float64},Measures.Length{:cy,Float64}},Rotation{Tuple{Measures.Length{:w,Float64},Measures.Length{:h,Float64}}},Tuple{Measures.Length{:mm,Float64},Measures.Length{:mm,Float64}}}}(Compose.TextPrimitive{Tuple{Measures.Length{:cx,Float64},Measures.Length{:cy,Float64}},Rotation{Tuple{Measures.Length{:w,Float64},Measures.Length{:h,Float64}}},Tuple{Measures.Length{:mm,Float64},Measures.Length{:mm,Float64}}}[Compose.TextPrimitive{Tuple{Measures.Length{:cx,Float64},Measures.Length{:cy,Float64}},Rotation{Tuple{Measures.Length{:w,Float64},Measures.Length{:h,Float64}}},Tuple{Measures.Length{:mm,Float64},Measures.Length{:mm,Float64}}}((0.5cx, 0.5cy), "plain text", Compose.HCenter(), Compose.VCenter(), Rotation{Tuple{Measures.Length{:w,Float64},Measures.Length{:h,Float64}}}(0.0, (0.5w, 0.5h)), (0.0mm, 0.0mm))], Symbol(""))]), List([Compose.Property{Compose.FontSizePrimitive}(Compose.FontSizePrimitive[Compose.FontSizePrimitive(7.055555555555555mm)]), Compose.Property{Compose.FontPrimitive}(Compose.FontPrimitive[Compose.FontPrimitive("Segoe UI")])]), 0, false, false, false, false, nothing, nothing, 0.0, Symbol(""))

julia> draw(PNG(), img_plain)
Compose.Image{Compose.PNGBackend}(IOBuffer(data=UInt8[...], readable=true, writable=true, seekable=true, append=false, size=2485, maxsize=Inf, ptr=2486, mark=-1), Cairo.CairoSurfaceBase{UInt32}(Ptr{Nothing} @0x0000000000000000, 535.0, 378.0), Cairo.CairoContext(Ptr{Nothing} @0x0000000000000000, Cairo.CairoSurfaceBase{UInt32}(Ptr{Nothing} @0x0000000000000000, 535.0, 378.0), Ptr{Nothing} @0x00007fd12118b5a0), 534.5059133378628, 377.9527559055118, RGBA{Float64}(0.0,0.0,0.0,0.0), RGBA{Float64}(0.0,0.0,0.0,1.0), Float64[], LineCapButt(), Compose.LineJoinMiter(), true, 0.3mm, 3.880555555555555mm, "Helvetica Neue,Helvetica,Arial,sans", nothing, false, Compose.ImagePropertyState[], Compose.ImagePropertyFrame[], Dict{Type,Union{Nothing, Compose.Property}}(), true, false, nothing, true, true, 3.7795275590551185, nothing, nothing)
false

julia> draw(PNG("plain.png"), img_plain)
```
plain

**Comment by CiaranOMara** [comment]
The example works on Linux 5.3.5-arch1-1-ARCH x86_64.

tmp

**Comment by CiaranOMara** [comment]
@Mattriks, your unicode example works for me on macOS 10.15 with https://github.com/GiovineItalia/Compose.jl/pull/360/commits/076581d81f17c6fab88ed699135c83a8b06c5b73 and  https://github.com/JuliaGraphics/Fontconfig.jl/pull/25, does this combination work for you on Windows 10?

**Comment by CiaranOMara** [comment]
Waiting for [Fontconfig v0.3](https://github.com/JuliaRegistries/General/pull/4330) to be registered.

**Comment by CiaranOMara** [comment]
Tests failed for Julia v0.7. Fontconfig v0.3 has a compatibility requirement of Julia ≥ v1. Dropping support for Julia v0.7 will allow a quick transition to a Binary Provider solution... how do you want to approach this?

**Comment by tlnagy** [comment]
https://github.com/GiovineItalia/Gadfly.jl/issues/1328 I think we're probably going to go ahead and drop 0.7 support given the difficulties we're running into with testing on it. We'll make a decision soon.

**Comment by bjarthur** [comment]
0.7 has been dropped.  https://github.com/GiovineItalia/Compose.jl/pull/362 .  redo CI and let's see if test pass now.

File: test/misc.jl [hunk 1]
```diff
@@ -157,8 +157,9 @@ end
 end
 
 @testset "Image keyword args" begin
-    @test typeof(PNG("foo.png", 4inch, 3inch, dpi=172)) <: Compose.Image
-    rm("foo.png")
+    mktemp() do tmp_path, tmp_io
+        @test typeof(PNG(tmp_path, 4inch, 3inch, dpi=172)) <: Compose.Image
```
Comment (Mattriks) [inline test/misc.jl]: If the point of this test is only to test the `dpi` keyword arg, then 
```julia
@test isa(PNG(4inch, 3inch, dpi=172), Compose.Image)
```
will work equally well.

Comment (CiaranOMara) [reply test/misc.jl]: That's much more concise.

File: .travis.yml [hunk 2]
```diff
@@ -2,29 +2,26 @@ language: julia
 os:
     - linux
     - osx
+    - windows
 sudo: required
 dist: trusty
 julia:
     - 1.0
+    - 1.1
```
Comment (bjarthur) [inline .travis.yml]: why test on 1.1?  1.0 is the long-term support, and 1.2 is the latest stable, and nightly is good of course.  i'd drop 1.1.

File: Project.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 name = "Compose"
 uuid = "a81c6b42-2e10-5240-aca2-a61377ecd94b"
-version = "0.7.4"
+version = "0.7.5"
```
Comment (bjarthur) [inline Project.toml]: the version bump commit should be after all of your other commits.  maybe do it in a separate PR?

File: .travis.yml (hunk 2 above)
Comment (CiaranOMara) [reply .travis.yml]: Greater visibility. I think there is little cost in checking that v1.1 is working as expected. 

I have removed v1.1 from the test matrix.

File: Project.toml (hunk 3 above)
Comment (CiaranOMara) [reply Project.toml]: I s'pose it is a bit eager. I have removed the version increment.

I'm keen to see Gadfly running/making use of BinaryProvider.
